syntax = "proto3";

package user_management;

service UserService {
  rpc RegisterUser (RegisterUserRequest) returns (RegisterUserResponse);
  rpc LoginUser (LoginUserRequest) returns (LoginUserResponse);
  rpc GetUserProfile (GetUserProfileRequest) returns (GetUserProfileResponse);
}

message RegisterUserRequest {
  string username = 1;
  string password = 2;
  string email = 3;
}

message RegisterUserResponse {
  bool success = 1;
  string message = 2;
}

message LoginUserRequest {
  string username = 1;
  string password = 2;
}

message LoginUserResponse {
  bool success = 1;
  string token = 2;
}

message GetUserProfileRequest {
  string token = 1;
}

message GetUserProfileResponse {
  string username = 1;
  string email = 2;
}

